pkg/image/prune/imagepruner.go

```diff
@@ -53,12 +67,59 @@ type ImagePruner interface {
 
 // imagePruner implements ImagePruner.
 type imagePruner struct {
-	g         graph.Graph
-	algorithm pruneAlgorithm
+	g              graph.Graph
+	algorithm      pruneAlgorithm
+	registryPinger registryPinger
 }
 
 var _ ImagePruner = &imagePruner{}
 
+// registryPinger performs a health check against a registry.
+type registryPinger interface {
+	// ping performs a health check against registry.
+	ping(registry string) error
+}
+
+// defaultRegistryPinger implements registryPinger.
+type defaultRegistryPinger struct {
+}
+
+func (drp *defaultRegistryPinger) ping(registry string) error {
+	healthzCheck := func(proto, registry string) error {
+		healthzResponse, err := http.Get(fmt.Sprintf("%s://%s/healthz", proto, registry))
```

Comment:
> Hrm - I'm confused as to what we are doing here.  What's the rationale
> behind a probe, vs failing the prune if the layers can't be removed?

If I recall correctly, since the information about exactly what needs to be removed in the registry is only being built in a forward direction and not stored, if we're unable to contact the registry the registry pieces will be orphaned.  Rather than orphan registry data with no hope of finding it, it seems better to fail early.